### PR TITLE
added spark cluster test to integration testing

### PR DESCRIPTION
**Pull request by Petro-Kotsiuba**

### CODE REVIEW

File: integration-tests/examples/ec2_templates/SparkStandalone.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "image" : "docker.dlab-dataengine",
+  "name" : "set the name",
+  "dataengine_instance_count" : "3",
+  "dataengine_slave" : "c4.large",
+  "dataengine_master" : "c4.large",
+  "notebook_name" : "set noteboork name",
+  "template_name" : "EMR cluster"
+}
```
Comment (YuriyHolinkoEp) [inline integration-tests/examples/ec2_templates/SparkStandalone.json]: It's not EMR. Please use "Spark cluster" template name

File: integration-tests/src/main/java/com/epam/dlab/automation/http/ApiPath.java [hunk 2]
```diff
@@ -27,6 +27,7 @@
     public static final String EXP_ENVIRONMENT = "/api/infrastructure_provision/exploratory_environment";
     public static final String PROVISIONED_RES = "/api/infrastructure/info";
     public static final String COMPUTATIONAL_RES = "/api/infrastructure_provision/computational_resources/dataengine-service";
+    public static final String COMPUTATIONAL_RES_STANDALONE = "/api/infrastructure_provision/computational_resources/dataengine";
```
Comment (YuriyHolinkoEp) [inline integration-tests/src/main/java/com/epam/dlab/automation/http/ApiPath.java]: COMPUTATIONAL_RES_STANDALONE. Pleas rename to COMPUTATIONAL_RES_SPARK

File: integration-tests/src/main/java/com/epam/dlab/automation/model/DeploySparkStandaloneDto.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+	}
+	
+	@Override
+    public String toString() {
+        return MoreObjects.toStringHelper(this)
+        		.add("image", image)
+        		.add("template_name", template_name)
+        		.add("name", name)
+        		.add(dataengine_master, dataengine_master)
+        		.add(dataengine_slave, dataengine_slave)
```
Comment (YuriyHolinkoEp) [inline integration-tests/src/main/java/com/epam/dlab/automation/model/DeploySparkStandaloneDto.java]: Strange toString(). Use key strings to identify each property

File: integration-tests/src/test/java/com/epam/dlab/automation/test/TestCallable.java [hunk 4]
```diff
@@ -77,10 +82,36 @@ public TestCallable(String notebookTemplate, boolean fullTest) {
...
+        switch (dataEngineType) {
+		case "dataengine":
+			testEMR();
+			break;
+		case "dataengine-service":
+			testSparkStandalone();
+			break;
+		default:
+			LOGGER.error("illegal dataengine type {}",dataEngineType);
+			fail("illegal dataengine type"+dataEngineType, new IllegalArgumentException("illegal dataengine type"+dataEngineType));
```
Comment (YuriyHolinkoEp) [inline integration-tests/src/test/java/com/epam/dlab/automation/test/TestCallable.java]: fail() methods throws exception with a message its own. Do not create a new exception object

File: integration-tests/src/test/java/com/epam/dlab/automation/test/TestCallable.java [hunk 5]
```diff
@@ -99,18 +130,90 @@ public Boolean call() throws Exception {
...
+       	terminateNotebook(dSparkStandaloneDto);
+       }
+	}
+
+  
+
+private DeploySparkStandaloneDto createStandalone() throws Exception {
+	// TODO Auto-generated method stub
+	String gettingStatus;
+    LOGGER.info("7. SparkStamdalone {} will be deployed for {} ...", emrName, notebookName);
```
Comment (YuriyHolinkoEp) [inline integration-tests/src/test/java/com/epam/dlab/automation/test/TestCallable.java]: Spelling

File: integration-tests/src/test/java/com/epam/dlab/automation/test/TestCallable.java [hunk 6]
```diff
@@ -99,18 +130,90 @@ public Boolean call() throws Exception {
         if (deployEMR != null) {
         	terminateNotebook(deployEMR);
         }
+	}
+
+   private void testSparkStandalone() throws Exception {
+		// TODO Auto-generated method stub
+	   final String notebookIp = createNotebook(notebookName);
```
Comment (YuriyHolinkoEp) [inline integration-tests/src/test/java/com/epam/dlab/automation/test/TestCallable.java]: What you need TODO?

File: integration-tests/src/test/java/com/epam/dlab/automation/test/TestCallable.java [hunk 7]
```diff
@@ -77,10 +82,36 @@ public TestCallable(String notebookTemplate, boolean fullTest) {
...
 
-        final DeployEMRDto deployEMR = createEMR();
+       	LOGGER.info("{} All tests finished successfully", notebookName);
+        return true;
+   }
+    
+    private void testEMR() throws Exception {
+		// TODO Auto-generated method stub
+    	final String notebookIp = createNotebook(notebookName);
+        testLibs();
```
Comment (YuriyHolinkoEp) [inline integration-tests/src/test/java/com/epam/dlab/automation/test/TestCallable.java]: final String notebookIp = createNotebook(notebookName);
testLibs();

it's common lines and there is not difference between emr and spark so move out those lines from methods that test EMR or Spark

File: integration-tests/src/test/java/com/epam/dlab/automation/test/TestCallable.java [hunk 8]
```diff
@@ -99,18 +130,90 @@ public Boolean call() throws Exception {
         if (deployEMR != null) {
         	terminateNotebook(deployEMR);
         }
+	}
+
+   private void testSparkStandalone() throws Exception {
+		// TODO Auto-generated method stub
+	   final String notebookIp = createNotebook(notebookName);
+       testLibs();
+//   	final DeployEMRDto deployEMR = createEMR();
```
Comment (YuriyHolinkoEp) [inline integration-tests/src/test/java/com/epam/dlab/automation/test/TestCallable.java]: Why you need this comment?

File: integration-tests/src/test/java/com/epam/dlab/automation/test/TestCallable.java [hunk 9]
```diff
@@ -99,18 +130,90 @@ public Boolean call() throws Exception {
...
+    LOGGER.info("{}: SparkStamdalone = {}",notebookName, sparkStandaloneDto);
+    Response responseDeployingSparkStandalone = new HttpRequest().webApiPut(ssnCompResURL, ContentType.JSON,
+    		sparkStandaloneDto, token);
+    LOGGER.info("{}:   responseDeployingSparkStandalone.getBody() is {}",notebookName, responseDeployingSparkStandalone.getBody().asString());
+    Assert.assertEquals(responseDeployingSparkStandalone.statusCode(), HttpStatusCode.OK, "SparkStamdalone  " + emrName + " was not deployed");
+
+    gettingStatus = WaitForStatus.emr(ssnProUserResURL, token, notebookName, emrName, "creating", ConfigPropertyValue.getTimeoutEMRCreate());
+    if(!ConfigPropertyValue.isRunModeLocal()) {
+        if (!(gettingStatus.contains("configuring") || gettingStatus.contains("running")))
+            throw new Exception(notebookName + ": SparkStamdalone " + emrName + " has not been deployed. SparkStamdalone status is " + gettingStatus);
```
Comment (YuriyHolinkoEp) [inline integration-tests/src/test/java/com/epam/dlab/automation/test/TestCallable.java]: Spelling

File: integration-tests/src/main/java/com/epam/dlab/automation/helper/NamingHelper.java [hunk 10]
```diff
@@ -91,7 +91,10 @@ public static String getNotebookInstanceName(String notebookName) {
     }
     
     public static String getClusterInstanceName(String notebookName, String clusterName, String dataEngineType) {
-    	return String.join("-", serviceBaseName, ConfigPropertyValue.getUsernameSimple(), dataEngineType, notebookName, clusterName);
+    	if("dataengine".equals(dataEngineType)) 
+    		return String.join("-", serviceBaseName, ConfigPropertyValue.getUsernameSimple(), dataEngineType, notebookName, clusterName, "master");
+    	else
```
Comment (YuriyHolinkoEp) [inline integration-tests/src/main/java/com/epam/dlab/automation/helper/NamingHelper.java]: braces {}

File: integration-tests/src/main/java/com/epam/dlab/automation/helper/NamingHelper.java [hunk 11]
```diff
@@ -112,8 +112,7 @@ public static String generateRandomValue() {
     }
 
     public static String generateRandomValue(String notebokTemplateName) {
-        SimpleDateFormat df = new SimpleDateFormat("YYYYMMddhmmss");
-        return String.join("_",  "ITest", notebokTemplateName, df.format(new Date()), String.valueOf(idCounter.incrementAndGet()));
+        return String.join("_",  "ITest", notebokTemplateName, String.valueOf(idCounter.incrementAndGet()));
```
Comment (YuriyHolinkoEp) [inline integration-tests/src/main/java/com/epam/dlab/automation/helper/NamingHelper.java]: get rid of "ITest"

File: integration-tests/src/main/java/com/epam/dlab/automation/helper/NamingHelper.java [hunk 12]
```diff
@@ -90,16 +90,19 @@ public static String getNotebookInstanceName(String notebookName) {
     	return String.join("-", serviceBaseName, ConfigPropertyValue.getUsernameSimple(), "nb", notebookName);
     }
     
-    public static String getEmrInstanceName(String notebookName, String emrName) {
-    	return String.join("-", serviceBaseName, ConfigPropertyValue.getUsernameSimple(), "emr", notebookName, emrName);
+    public static String getClusterInstanceName(String notebookName, String clusterName, String dataEngineType) {
+    	if("dataengine".equals(dataEngineType)) 
+    		return String.join("-", serviceBaseName, ConfigPropertyValue.getUsernameSimple(), "de", notebookName, clusterName, "m");
```
Comment (YuriyHolinkoEp) [inline integration-tests/src/main/java/com/epam/dlab/automation/helper/NamingHelper.java]: braces

File: integration-tests/src/main/java/com/epam/dlab/automation/helper/NamingHelper.java (hunk 11 above)
Comment (Petro-Kotsiuba) [reply integration-tests/src/main/java/com/epam/dlab/automation/helper/NamingHelper.java]: done

File: integration-tests/src/main/java/com/epam/dlab/automation/helper/NamingHelper.java (hunk 12 above)
Comment (Petro-Kotsiuba) [reply integration-tests/src/main/java/com/epam/dlab/automation/helper/NamingHelper.java]: done

**Comment by YuriyHolinkoEp** [comment]
@Petro-Kotsiuba Good!!! Waiting for result of integration tests

File: integration-tests/src/test/java/com/epam/dlab/automation/test/TestServices.java [hunk 13]
```diff
@@ -68,12 +71,18 @@
 	public static final int N_THREADS = 10;
 
 	private long testTimeMillis;
+	private ArrayList<NotebookConfig> notebookConfigs;
```
Comment (YuriyHolinkoEp) [inline integration-tests/src/test/java/com/epam/dlab/automation/test/TestServices.java]: List<>

File: integration-tests/src/test/java/com/epam/dlab/automation/test/TestServices.java [hunk 14]
```diff
@@ -68,12 +71,18 @@
...
 
 	@BeforeClass
-	public void Setup() throws InterruptedException {
+	public void Setup() throws InterruptedException, JsonParseException, JsonMappingException, IOException {
```
Comment (YuriyHolinkoEp) [inline integration-tests/src/test/java/com/epam/dlab/automation/test/TestServices.java]: JsonParseException, JsonMappingException, IOException 
it's all about IOException

File: integration-tests/examples/ec2_templates/rstudio/spark_cluster.json [hunk 15]
```diff
@@ -0,0 +1,9 @@
+{
+  "image" : "docker.dlab-dataengine",
+  "name" : "set the name",
+  "dataengine_instance_count" : "2",
+  "dataengine_slave" : "p2.xlarge",
```
Comment (YuriyHolinkoEp) [inline integration-tests/examples/ec2_templates/rstudio/spark_cluster.json]: c4.xlarge for master and slave

File: integration-tests/examples/ec2_templates/zeppelin/spark_cluster.json [hunk 16]
```diff
@@ -0,0 +1,9 @@
+{
+  "image" : "docker.dlab-dataengine",
+  "name" : "set the name",
+  "dataengine_instance_count" : "2",
+  "dataengine_slave" : "p2.xlarge",
```
Comment (YuriyHolinkoEp) [inline integration-tests/examples/ec2_templates/zeppelin/spark_cluster.json]: c4.xlarge for master and slave

File: integration-tests/examples/ec2_templates/jupyter/spark_cluster.json [hunk 17]
```diff
@@ -0,0 +1,9 @@
+{
+  "image" : "docker.dlab-dataengine",
+  "name" : "set the name",
+  "dataengine_instance_count" : "2",
+  "dataengine_slave" : "p2.xlarge",
```
Comment (YuriyHolinkoEp) [inline integration-tests/examples/ec2_templates/jupyter/spark_cluster.json]: c4.xlarge for master and slave

**Pull request merged**